test/Geometry/GeometryTestAlg.cxx

```diff
@@ -2355,61 +2390,103 @@ namespace geo{
      */
 
     unsigned int nErrors = 0;
-    for (geo::GeometryCore::TPC_id_iterator iTPC(&*geom); iTPC; ++iTPC) {
-      const geo::TPCGeo& TPC = *(iTPC.get());
+    for (geo::TPCGeo const& TPC: geom->IterateTPCs()) {
 
-      MF_LOG_DEBUG("GeometryTest") << "Cryostat #" << iTPC->Cryostat
-        << " TPC #" << iTPC->TPC;
+      MF_LOG_DEBUG("GeometryTest") << "Wire intersection test on " << TPC.ID();
 
       // sanity: wires on different cryostats
-      if (iTPC->Cryostat < geom->Ncryostats() - 1) {
-        geo::WireID w1 { iTPC->Cryostat, iTPC->TPC, 0, 0 },
-          w2 { iTPC->Cryostat + 1, iTPC->TPC, 1, 1 };
-        geo::Point_t xingPoint;
-        if (geom->WireIDsIntersect(w1, w2, xingPoint)) {
-          MF_LOG_ERROR("GeometryTest") << "WireIDsIntersect() on " << w1
-            << " and " << w2 << " returned " << xingPoint
-            << " cm, while should have reported no intersection at all";
-          ++nErrors;
-        } // if intersect
-        try {
-          // the value of result is not checked here
-          geom->Wire(w1).IntersectionWith(geom->Wire(w2));
+      if (TPC.ID().Cryostat < geom->Ncryostats() - 1) {
+        
+        geo::WireID const w1 { geo::PlaneID{ TPC.ID(), 0 }, 0 };
+        geo::WireGeo const& wire1 = geom->Wire(w1);
+        geo::Vector_t const& wireDir = wire1.Direction<geo::Vector_t>();
+        
+        geo::CryostatGeo const& otherCryo
+          = geom->Cryostat(TPC.ID().Cryostat + 1);
+        geo::PlaneGeo const* otherPlane = nullptr;
+        for (geo::PlaneGeo const& plane: geom->IteratePlanes(otherCryo.ID())) {
+          if (planeAlignmentTo(plane, wireDir) != 0) continue;
+          otherPlane = &plane;
+          break;
         }
-        catch(...) {
-          MF_LOG_ERROR("GeometryTest") << "WiresIntersect() on " << w1
-            << " and " << w2 << " threw an exception, which should not have";
-          ++nErrors;
+        if (otherPlane) {
+          geo::WireID const w2 { otherPlane->ID(), 1 };
+          MF_LOG_TRACE("GeometryTest")
+            << "Off cryostat test (" << w1 << "): chosen wire " << w2;
+          geo::Point_t xingPoint;
+          if (geom->WireIDsIntersect(w1, w2, xingPoint)) {
+            MF_LOG_ERROR("GeometryTest") << "WireIDsIntersect() on " << w1
+              << " and " << w2 << " returned " << xingPoint
+              << " cm, while should have reported no intersection at all";
+            ++nErrors;
+          } // if intersect
+          try {
+            // the value of result is not checked here
+            wire1.IntersectionWith(geom->Wire(w2));
+          }
+          catch(...) {
```

Comment:
do you know what the expected exception is supposed to be here?  If so it might be more informative to only catch that.

Reply:
There is no expected exception, which is kind of the reason why I backed up to the hideous `catch(...)`: but, related to your question, currently the implementation does not explicitly throw exceptions.
My rationalisation is the following. I have two options: either catch any exception, or catch none. The goal of this test is to detect and report all issues encountered, so that they can be debugged afterwards. So it seems more on target to have the code trigger the most possible errors, record them, and at the end report all of them.

I am still fought on this: I myself would jump on anybody using such constructs in regular code, and now I am going outside of the standard rules because I need to _test_ non-standard situations. So I am still open to changes if my arguments are not satisfactory.

Reply:
The only reason I suggest this is because you're potentially hiding other errors by attempting to catch everything.  If you truly don't know what the exception could be then this is your only try/catch option.  I guess you could make the printout more verbose and print what the exception is?

Reply:
As far as I know, C++ does not allow to act on the exception caught by `catch (...)`.
The thing that goes closer is to catch a `std::exception` hoping that whatever is caught derives from it, and then you can get both the type of the exception and `what()` it is objecting about; and then a catch-all as fall back.
Hmm... maybe that's not a bad idea after all.